### ISSUE TITLE
Provide a Bzlmod module to support Bazel6

### ISSUE THREAD

**Issue by opicaud** [protocolbuffers/protobuf#13618]
**What language does this apply to?**
If it's a proto syntax change, is it for proto2 or proto3? No
If it's about generated code change, what programming language? No

**Describe the problem you are trying to solve.**
Being able to use protobuf as module in a Bazel project

**Describe the solution you'd like**
Providing a MODULE.bazel file

**Describe alternatives you've considered**

**Additional context**
There is a draft here #12489 but not updated for April 2023, maybe @deannagarcia ?

If the request sounds valuable, i can contribute to provide the module file
Olivier

**Comment by meteorcloudy** [comment]
FYI @comius

**Comment by aaliddell** [comment]
Still desired

**Comment by bgdnvk** [comment]
running into this issue with gazelle and protobuffs deps

**Comment by railwaycat** [comment]
There are already patches under [Bazel Central Registry](https://github.com/bazelbuild/bazel-central-registry/tree/main/modules/protobuf) to bring bzlmod support to protobuf. But only for several history versions. Is it possible to reuse those patches for the latest code?

**Comment by udf2457** [comment]
ping

**Comment by udf2457** [comment]
$hello_world

**Comment by meteorcloudy** [comment]
protobuf is already published to https://registry.bazel.build/modules/protobuf and managed by the protobuf team.

**Comment by touchinglie** [comment]
I'm installing the ProtoBuf lib on my Radxa Rock 3A Debian Bullseye to compile the source code for NCNN.
When I executed the command,

bazel build :protoc :protobuf

it raise an error:

INFO: Reading rc options for 'build' from /home/radxa/protobuf-28.1/.bazelrc: Inherited 'common' options: --noenable_bzlmod --repo_env=BAZEL_NO_APPLE_CPP_TOOLCHAIN=1 ERROR: --noenable_bzlmod :: Unrecognized option: --noenable_bzlmod

I checked .bazelrc, and found this link. 
Any Idea or help pls.
